### PR TITLE
🍰 2019/kw20/fix_lint_rules_and_vscode

### PR DESCRIPTION
**Pull request by Tirokk**
> [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-18T13:49:23Z" title="Saturday, May 18th 2019, 3:49:23 pm +02:00">May 18, 2019</time>_
_Merged <time datetime="2019-05-23T18:22:34Z" title="Thursday, May 23rd 2019, 8:22:34 pm +02:00">May 23, 2019</time>_
---

## 🍰 Pullrequest

- Unified lint rules (duplicate definitions)
- more lint rules
- prettier lets lint fail
- vscode settings
- vscode recommended plugins

### Issues

- [X] fixes https://github.com/Human-Connection/Human-Connection/issues/590

### Todo

- [x] Are the rules ok? @roschaefer 
- [x] Autofix lint problems
- [x] Fix remaining problems (64 frontend & 5 backend)

### CODE REVIEW

**Comment by Tirokk** [comment]
> [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-20T09:03:39Z" title="Monday, May 20th 2019, 11:03:39 am +02:00">May 20, 2019</time>_

---


@roschaefer I did not fix the lint errors so you can review. I plan to merge this at the end of the week after all other PRs we can merge are merged in oder to minimize conflicts

**Comment by Tirokk** [comment]
> [codecov[bot]](https://github.com/apps/codecov) **Authored by [codecov[bot]](https://github.com/apps/codecov)**
_<time datetime="2019-05-23T18:21:26Z" title="Thursday, May 23rd 2019, 8:21:26 pm +02:00">May 23, 2019</time>_

---


# [Codecov](https://codecov.io/gh/Human-Connection/Human-Connection/pull/625?src=pr&el=h1) Report
> Merging [#625](https://codecov.io/gh/Human-Connection/Human-Connection/pull/625?src=pr&el=desc) into [master](https://codecov.io/gh/Human-Connection/Human-Connection/commit/99ae9770c67b8c2697f1d53e75c65880cb00cc6e?src=pr&el=desc) will **increase** coverage by `0.05%`.
> The diff coverage is `17.86%`.

[Impacted file tree graph](https://codecov.io/gh/Human-Connection/Human-Connection/pull/625?src=pr&el=tree)

```diff
@@            Coverage Diff            @@
##           master    #625      +/-   ##
=========================================
+ Coverage   20.14%   20.2%   +0.05%     
=========================================
  Files         157     157              
  Lines        2209    2208       -1     
  Branches      105     105              
=========================================
+ Hits          445     446       +1     
+ Misses       1740    1738       -2     
  Partials       24      24
```


| [Impacted Files](https://codecov.io/gh/Human-Connection/Human-Connection/pull/625?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [webapp/pages/settings.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/625/diff?src=pr&el=tree#diff-d2ViYXBwL3BhZ2VzL3NldHRpbmdzLnZ1ZQ==) | `0% <ø> (ø)` | :arrow_up: |
| [backend/src/middleware/activityPubMiddleware.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/625/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvbWlkZGxld2FyZS9hY3Rpdml0eVB1Yk1pZGRsZXdhcmUuanM=) | `0% <ø> (ø)` | :arrow_up: |
| [webapp/pages/post/\_id.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/625/diff?src=pr&el=tree#diff-d2ViYXBwL3BhZ2VzL3Bvc3QvX2lkLnZ1ZQ==) | `0% <ø> (ø)` | :arrow_up: |
| [webapp/components/Empty.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/625/diff?src=pr&el=tree#diff-d2ViYXBwL2NvbXBvbmVudHMvRW1wdHkudnVl) | `100% <ø> (ø)` | :arrow_up: |
| [...omponents/notifications/NotificationMenu/index.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/625/diff?src=pr&el=tree#diff-d2ViYXBwL2NvbXBvbmVudHMvbm90aWZpY2F0aW9ucy9Ob3RpZmljYXRpb25NZW51L2luZGV4LnZ1ZQ==) | `30% <ø> (ø)` | :arrow_up: |
| [webapp/components/Modal.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/625/diff?src=pr&el=tree#diff-d2ViYXBwL2NvbXBvbmVudHMvTW9kYWwudnVl) | `76.92% <ø> (ø)` | :arrow_up: |
| [webapp/pages/moderation.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/625/diff?src=pr&el=tree#diff-d2ViYXBwL3BhZ2VzL21vZGVyYXRpb24udnVl) | `0% <ø> (ø)` | :arrow_up: |
| [webapp/components/Modal/DisableModal.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/625/diff?src=pr&el=tree#diff-d2ViYXBwL2NvbXBvbmVudHMvTW9kYWwvRGlzYWJsZU1vZGFsLnZ1ZQ==) | `93.33% <ø> (ø)` | :arrow_up: |
| [webapp/components/User/index.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/625/diff?src=pr&el=tree#diff-d2ViYXBwL2NvbXBvbmVudHMvVXNlci9pbmRleC52dWU=) | `92.31% <ø> (ø)` | :arrow_up: |
| [webapp/components/Editor/plugins/eventHandler.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/625/diff?src=pr&el=tree#diff-d2ViYXBwL2NvbXBvbmVudHMvRWRpdG9yL3BsdWdpbnMvZXZlbnRIYW5kbGVyLmpz) | `20% <ø> (ø)` | :arrow_up: |
| ... and [137 more](https://codecov.io/gh/Human-Connection/Human-Connection/pull/625/diff?src=pr&el=tree-more) | |

**Review by Mogge — APPROVED** [review]
Looks good to me :+1: could you fix the lint errors on Travis?

Would be good if we could also update prettier https://github.com/Human-Connection/Human-Connection/pull/250

**Pull request closed without merging**
